src/controllers/imgurController.js

```diff
@@ -0,0 +1,68 @@
+const { ImgurClient } = require('imgur');
+const User = require('../models/User');
+
+const imgurClient = new ImgurClient({
+  clientId: process.env.IMGUR_CLIENT_ID,
+  clientSecret: process.env.IMGUR_CLIENT_SECRET,
+  refreshToken: process.env.IMGUR_REFRESH_TOKEN,
+});
+
+const renderUploadPage = (req, res) => {
+  res.render('imgur');
+};
+
+async function uploadProfile(req, res) {
+  console.log('Received request for /api/v1/user/:userId/uploadProfile');
+  const userId = req.params.userId;
+  console.log('User ID:', userId);
+
+  if (!userId) {
+    return res.status(400).send('No user id was provided.');
+  }
+
+  if (!req.files || Object.keys(req.files).length === 0) {
+    console.log('No files were uploaded.');
+    return res.status(400).send('No files were uploaded.');
+  }
+
+  const user = await User.findById(userId);
+  console.log('User:', user);
+  if (!user) {
+    return res.status(400).send('No user found');
+  }
+
+  const profile = req.files.profile;
+  console.log('File Details:', profile);
+
+  if (!profile) {
+    console.log('No profile file in the request.');
+    return res.status(400).send('No profile file in the request.');
+  }
+  console.log('File MIME Type:', profile.mimetype);
+  console.log('File Size:', profile.size);
+
+  if (!/^image/.test(profile.mimetype)) {
+    return res.status(400).send('File is not an image.');
+  }
+
+  const imgurRes = await imgurClient.upload({
+    image: profile.data.toString('base64'),
+    type: 'base64',
+    title: `${user._id}_profile_pic`,
+  });
+
+  if (imgurRes.success) {
+    user.profileImage = {
+      url: imgurRes.data.link,
+      altText: 'user profile picture',
+    };
+    await user.save();
+  }
+
+  res.render('imgur', { userId: req.params.userId });
```

Comment:
It might be nice to pass in a variable like `message`, so that we can show when the page reloads some text that says "Image uploaded successfully" or something. Or you could even pass back the image link so that we can show the image on the page once successful :)

Reply:
Another idea, if you feel like trying your hand at it; is to return a JSON object here instead of re-rendering the page.

Then, in the `imgur.ejs` file you would add a `<script>` tag. 

Then in the `src/public` folder you would create a javascript file that the `script` tag should poin tto . This javascript file is where you would handle submitting the form (instead of letting the default DOM behaviour handle it like you have right now). 

[https://matiashernandez.dev/blog/post/handling-forms-with-vanilla-javascript](https://matiashernandez.dev/blog/post/handling-forms-with-vanilla-javascript)

The main advantage of that is that you can add custom logic, like showing a Toast/Alert bar when the request is successful; or dynamically showing the uploaded image along with a message. If you want to see an example of showing a toast you can look at the orderNotification page: [https://github.com/Code-the-Dream-School/dd-prac-team4-back/blob/7c9b523158abe7f8e952de14b3bbdca95d4584cf/src/public/orderNotifications.js#L20-L31](https://github.com/Code-the-Dream-School/dd-prac-team4-back/blob/7c9b523158abe7f8e952de14b3bbdca95d4584cf/src/public/orderNotifications.js#L20-L31)

If you want to see an example of submitting a form via javascript, you can look at the admin album update page: https://github.com/Code-the-Dream-School/dd-prac-team4-back/blob/7c9b523158abe7f8e952de14b3bbdca95d4584cf/src/public/admin.js#L7

Reply:
Thank you very much for your suggestions 👍, as for the message and image link , implemented in this commit 676f52ad0aef21e29a81f7d3703441773c9ad82c

Reply:
> Another idea, if you feel like trying your hand at it; is to return a JSON object here instead of re-rendering the page.
> 
> Then, in the `imgur.ejs` file you would add a `<script>` tag. 
> 
> Then in the `src/public` folder you would create a javascript file that the `script` tag should poin tto . This javascript file is where you would handle submitting the form (instead of letting the default DOM behaviour handle it like you have right now). 
> 
> https://matiashernandez.dev/blog/post/handling-forms-with-vanilla-javascript
> 
> The main advantage of that is that you can add custom logic, like showing a Toast/Alert bar when the request is successful; or dynamically showing the uploaded image along with a message. If you want to see an example of showing a toast you can look at the orderNotification page:
> 
> https://github.com/Code-the-Dream-School/dd-prac-team4-back/blob/7c9b523158abe7f8e952de14b3bbdca95d4584cf/src/public/orderNotifications.js#L20-L31
> 
> If you want to see an example of submitting a form via javascript, you can look at the admin album update page:
> 
> https://github.com/Code-the-Dream-School/dd-prac-team4-back/blob/7c9b523158abe7f8e952de14b3bbdca95d4584cf/src/public/admin.js#L7

Akos could you please check my attempt to realize the suggestion regarding to  the implementation of  a JSON object  instead of re-rendering the page, adding custom logic , toastify?   commit 588889e6ebaa1a5e20188da1695e3e159ba2bc59